.github/workflows/maven.yml

```diff
@@ -0,0 +1,21 @@
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'corretto'
+          cache: maven
+      - name: Compile Project With Maven
+        run: mvn clean test-compile
```

Comment:
This is basically going to be just allowing us to run tests (which is fine for Pull requests) but it should allow us to publish snapshots/releases into Maven Central when there are pushes to the `master` branch.

So we need to be splitting this into 2 workflow files:

1. which merely runs `mvn clean test` on `pull_requests` and 
2. which runs `mvn clean deploy` on `master` branch pushes.

You can refer to some samples from https://github.com/cbeust/testng/tree/master/.github/workflows

Can you please help make that change as well, so that we can do this move completely to github actions from travis ? I been meaning to get to this for quite sometime, but keep losing track of it.

Reply:
Sure, I'll update this PR. Thanks for the detailed info, Ill take a look at the resources as well!